### PR TITLE
String preconditions

### PR DESCRIPTION
**Pull request by le6o**

### CODE REVIEW

File: src/Preconditions/StringPreconditions.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+     * @throws ValueTooSmall      if the limit is less than or equal to zero.
+     * @throws ArgumentNotAString if the subject is not a string.
+     * @throws StringTooLong      if the subject is longer than or equal to $limit.
+     *
+     * @return $this
+     */
+    public function isShorterThan($limit)
+    {
+        \Guardsman\check($limit)->isGreaterThan(0);
+        \Guardsman\check($this->getSubject())->isString();
```
Comment (cs278) [inline src/Preconditions/StringPreconditions.php]: Can you not just use `$this->isString();`?

File: src/Preconditions/StringPreconditions.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        if (mb_strlen($this->getSubject()) >= $limit) {
```
Comment (cs278) [inline src/Preconditions/StringPreconditions.php]: FYI `mb_strlen()` way return false, I assume under invalid multibyte sequences. You way want to note that these methods are dependant on the value of the mbstring internal encoding.

File: src/Preconditions/StringPreconditions.php [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+{
+    /** @return mixed */
+    abstract public function getSubject();
+
+    /**
+     * @throws ArgumentNotAString if the subject is not a string.
+     *
+     * @return $this
+     */
+    public function isString()
```
Comment (cs278) [inline src/Preconditions/StringPreconditions.php]: If this is exclusively dealing with multibyte strings maybe it should use [`mb_check_encoding()`](http://php.net/manual/en/function.mb-check-encoding.php).

File: tests/Preconditions/StringPreconditionsTest.php [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    public function testIsStringThrowsArgumentNotAString($subject)
+    {
+        \Guardsman\check($subject)->isString();
+    }
+
+    public function testIsShorterThan()
+    {
+        $this->assertInstanceOf(
+            \Guardsman\Guardsman::class,
+            \Guardsman\check('string')->isShorterThan(7)
```
Comment (cs278) [inline tests/Preconditions/StringPreconditionsTest.php]: Maybe test some unicode strings as well…

File: src/Preconditions/StringPreconditions.php (hunk 1 above)
Comment (le6o) [inline src/Preconditions/StringPreconditions.php]: Yeah true story, I think I wanted to demo the main syntax use of the library but that seems pointless now I think about it again.

**Pull request merged**